packages/manager/modules/vps/src/dashboard/tile/configuration/service.js

```diff
@@ -0,0 +1,164 @@
+import find from 'lodash/find';
+import get from 'lodash/get';
+import head from 'lodash/head';
+
+class ConfigurationTileService {
+  constructor() {
+    this.vps = null;
+    this.vpsModel = null;
+
+    this.availableUpgrades = null;
+  }
+
+  get currentPlan() {
+    return find(this.availableUpgrades, {
+      planCode: this.vpsModel.name,
+    });
+  }
+
+  static getPlanPriceDiff(upperPlan, currentPlan) {
+    const upperPlanTotalPriceValue = ConfigurationTileService.getPlanPriceValue(
+      upperPlan,
+    );
+    const currentPlanTotalPriceValue = ConfigurationTileService.getPlanPriceValue(
+      currentPlan,
+    );
+
+    return ConfigurationTileService.getPriceStructure(
+      upperPlanTotalPriceValue - currentPlanTotalPriceValue,
+      head(currentPlan.prices).price,
+    );
+  }
+
+  static getPlanPriceValue(plan) {
+    return plan.prices.reduce((total, { price }) => total + price.value, 0);
+  }
+
+  static getPriceStructure(value, { currencyCode, text }) {
+    return {
+      currencyCode,
+      text: text.replace(/\d+(?:[.,]\d+)?/, value.toFixed(2)),
+      value,
+    };
+  }
+
+  /**
+   *  Set the vps informations
+   *
+   *  @param {Object} vps      The informations provided by 2API
+   *  @param {Object} vpsModel The model attribute provided by
+   *                           GET /vps/{servicName} (not present in 2API...)
```

Comment:
```suggestion
   *                           GET /vps/{serviceName} (not present in 2API...)
```
'Not present in 2API' > Would it make sense to add it to avoid one more call ?

Reply:
The call is already done in fact :-)
And in my opinion the VPS information 2API is bad because it changes too much the response of the APIv6 call. The 2API should be rewritten in order to have something coherent.